libs/qCC_db/src/ccRasterGrid.cpp

```diff
@@ -244,6 +245,20 @@ bool ccRasterGrid::fillWith(	ccGenericPointCloud* cloud,
 	//we always handle the colors (if any)
 	hasColors = cloud->hasColors();
 
+	//Array of pointers, each coresponding to a point in the cloud
+	// cloud->getPoint(n)  coresponds to (pointRefList.data()+n)
+	// The pointers are used to chain together points, bellonging to the same cell 
+	std::vector<void*> pointRefList;
```

Comment:
I looked quicly at the code and there's something I dont understand with this pointRefList.

You create a vector with `pointCount` times `nullptr`.
Yet I dont see you modifying any of the pointers cointained inside this `pointRefList`

Reply:
It happens in:
*(aCell.pointRefTail) = pointRefList.data() + n;

It generates a linked list of points belonging to a cell
Each node is just a pointer to the next node belonging to the same cell

Normally, I would have had the next-pointer inside the struct that is to be linked, something like:
point{float x; float y; float z; point* next}

But I did not want to modify the data structure here, so instead I just added the linked pointers as a parallel array, sharing index with the point. If I had a next pointer in the CCVector3 structure, it would be:
cloud->getPoint(n)->next  instead of pointRefList[n]

The pointRef to the first to the point found in a given cell, is set to cell.pointRefHead
Then all following points are linked as a chain from this, keeping a reference to the last point appended in cell.pointRefTail